src/common/plugin.ts

```diff
@@ -47,10 +55,54 @@ export abstract class Plugin implements IPlugin {
 
     const response = this.container.channelService.hasPermission(channelName, this.permission);
     if (!response) {
-      message.reply(`Please use this command in a \`${this.permission}\` channel`);
+      const rooms = Object.values(Constants.Channels[this.permission]).splice(0, 3);
+      let addonText = '';
+
+      // not sure if this will cause issues as I do pop id, please let me know
+      const chanInfo: CChannelInfo[] = rooms
+        .filter((room) => {
+          return this.container.guildService
+            .getChannel(room)
+            .permissionsFor(message.member || '')
+            ?.has('VIEW_CHANNEL');
+        })
+        .map((room) => new CChannelInfo(this.container.guildService.getChannel(room).id, room))
+        .splice(0, 2);
```

Comment:
You spliced twice, remove the first splice on line 58 and only do it here, after the filter

Reply:
I splice twice for a reason, the first on is so we know if we have 2 or more channels or not.  The other one is because only the first 2 channels are needed beyond that point.

Reply:
To build up on this I did not want to just on that last one as the map would be doing pointless work at that point.

Reply:
Actually changed how I write the message so this is actually fixed now.